javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -289,29 +289,32 @@ int size() {
         }
     }
 
-    static <T, U extends Iterable<? extends Iterable<T>>> U transpose(U source, Function<Iterable<?>, Iterable<?>> mapper) {
-        if (source.iterator().hasNext()) {
-            return transposeIterables(source, mapper);
-        } else {
+    static <T, U extends Seq<T>, V extends Seq<U>> V transpose(V source, Function<Iterable<U>, V> outerMapper, Function<Iterable<T>, U> innerMapper) {
```

Comment:
I agree that it's minor.
Though `map` functions have a [`mapper` function param also](https://github.com/javaslang/javaslang/blob/master/javaslang/src/main/java/javaslang/collection/List.java#L896), factory would be too general, in my opinion.